contracts/contracts/Umbra.sol

```diff
@@ -81,21 +100,40 @@ contract Umbra is Ownable, ReentrancyGuard {
         )
         public
         payable
-        nonReentrant
     {
         require(msg.value == toll, "Umbra: Must pay the exact toll");
 
+        payments[_receiver] = Payment({token: _tokenAddr, amount: _amount});
         emit Announcement(_receiver, _amount, _tokenAddr, _iv, _pkx, _pky, _ct0, _ct1, _ct2, _mac);
 
-        SafeERC20.safeTransferFrom(IERC20(_tokenAddr), msg.sender, _receiver, _amount);
+        SafeERC20.safeTransferFrom(IERC20(_tokenAddr), _msgSender(), address(this), _amount);
+    }
+
+    function withdrawToken(address _acceptor) public {
+        uint256 amount = payments[_msgSender()].amount;
+        address tokenAddr = payments[_msgSender()].token;
+
+        require(
+            (amount > 0) && (tokenAddr != address(0)) && (tokenAddr != ETH_TOKEN_PLACHOLDER),
+            "Umbra: No tokens available for withdrawl"
```

Comment:
Thinking about this more, since we no longer hold ETH payments in the contract itself, this check isn't necessary, since there will never be an entry in the `payments` map for ETH payments, the first check `amount > 0` should always fail.

What this made me realize was that we should probably rename some stuff in the contract to reflect that only token payments would be stored here.

Reply:
Good point! Looks like you did those renames so I'll resolve this one